src/ripple/protocol/STObject.h

```diff
@@ -401,7 +401,7 @@ class STObject
         if (!cf)
             throw std::runtime_error ("Wrong field type");
 
-        cf->setValue (value);
+        cf->setValue (std::move (value));
```

Comment:
I'm still a bit worried.  Thanks for doing the due diligence to look at all the callers.  In the interests of future proofing, I'd feel a bit more comfortable if the body of the method held something like this:
```
static_assert (std::is_reference<V>::value == false, "setFieldUsingValue is not designed for references.");
```
Am I missing something?